File: hsg12-tinh/2023-2024-vinhphuc/expfuel/expfuel.old.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#define input_file "expfuel.inp"
#define output_file "expfuel.out"

using namespace std;

int n, m, c;
vector<pair<int, int>> A;
vector<int> D;
vector<int> result;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> n >> m >> c;

    A.resize(n);

    for (int i = 0; i < n; ++i)
    {
        cin >> A[i].first >> A[i].second;
    }

    D.resize(m);

    for (int i = 0; i < m; ++i)
    {
        cin >> D[i];
    }
}


// Hàm tính tổng nhiên liệu bao gồm đường bay và đường bộ
int fuel(int flying_fuel, int ai, int dj)
{
    return flying_fuel + c * abs(ai - dj);
}


void process()
{
    int min_fuel;

    // Duyệt điểm thám hiểm dj của các đội
    for (int j = 0; j < m; ++j)
    {
        min_fuel = INT_MAX; 

        // Duyệt các điểm tập kết a_i của đội dj đang xét
        for (const pair<int, int>& a : A)
        {
            min_fuel = min(min_fuel, fuel(a.second, a.first, D[j]));
        }

        result.push_back(min_fuel);    
    }
}


void output()
{
    freopen(output_file, "w", stdout);

    for (int j = 0; j < m; ++j)
    {
        cout << result[j] << '\n';
    }
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg12-tinh/2023-2024-vinhphuc/expfuel/expfuel.old.py
import os
import sys

input_file = os.path.join(sys.path[0], 'expfuel.inp')
output_file = os.path.join(sys.path[0], 'expfuel.out')

n, m, c = 0, 0, 0
A = []
D = []
result = []


def input_data():
    global n, m, c, A, D

    with open(input_file, 'r') as f:
        n, m, c = map(int, f.readline().split())

        for _ in range(n):
            position, fuel = map(int, f.readline().split())
            A.append((position, fuel))
        
        for _ in range(m):
            dj = int(f.readline())
            D.append(dj)


# Hàm tính tổng nhiên liệu bao gồm đường bay và đường bộ
def fuel(flying_fuel, ai, dj):
    return flying_fuel + c * abs(ai - dj)


def process():
    global A, D, result

    # Duyệt điểm thám hiểm dj của các đội
    for dj in D:
        min_fuel = float('inf')

        # Duyệt các điểm tập kết a_i của đội dj đang xét
        for position, flying_fuel in A:
            min_fuel = min(min_fuel, fuel(flying_fuel, position, dj))

        result.append(min_fuel)


def output():
    global result
    with open(output_file, 'w') as f:
        f.write('\n'.join(map(str, result)))


if __name__ == '__main__':
    input_data()
    process()
    output()